src/Controls/src/Core/Handlers/Items/Android/Adapters/ItemsViewAdapter.cs

```diff
@@ -95,7 +96,16 @@ public override int GetItemViewType(int position)
 		{
 			if (_usingItemTemplate)
 			{
-				return ItemViewType.TemplatedItem;
+				// If we're using a template selector, using the 
+				// selected DataTemplate's 'Type'
+				// and get the hashcode to get a reasonably unique
+				// value for the DataTemplate for us to recycle
+				// all views using this DataTemplate with
+				return _itemTemplateSelector?
```

Comment:
Until you make OnCreateViewHolder and OnBindViewHolder aware of the `viewType` integers you're generating, none of this is going to have any effect. 

GetHashCode for the value is a little... icky. Technically it will work, because even if the multiple template types end up having the same hash code, TemplatedItemViewHolder will catch it during Bind and fix it up. 

Actually, it's going to fail pretty spectacularly if the hash code happens to collide with one of the known view types in ItemViewType. E.g., if your CV has a templated header, and if the hash code happens to be 43 for your ItemTemplate, you're going to get items that look like your header. 

So at the very least, you need to shift the hash code result out of the range of stuff in ItemViewType. But even then, you're always at risk of hash code collisions between data templates causing the same template thrash that you're trying to avoid.